tests/container_collection/manifest/test_filter_manifest_files.py

```diff
@@ -0,0 +1,641 @@
+                "s3://bucket-name",
+                "full/file/name_key_A_0000.extension_b",
+            ],
+            [
+                "name_key_A_0001",
+                "extension_a",
+                "s3://bucket-name",
+                "full/file/name_key_A_0001.extension_a",
+            ],
+            [
+                "name_key_A_0001",
+                "extension_b",
+                "s3://bucket-name",
+                "full/file/name_key_A_0001.extension_b",
+            ],
+            [
+                "name_key_B_0000",
+                "extension_a",
+                "/local/path/",
+                "full/file/name_key_B_0000.extension_a",
+            ],
+            [
+                "name_key_B_0000",
+                "extension_b",
+                "/local/path/",
+                "full/file/name_key_B_0000.extension_b",
+            ],
+            [
+                "name_key_B_0001",
+                "extension_a",
+                "/local/path/",
+                "full/file/name_key_B_0001.extension_a",
+            ],
+            [
+                "name_key_B_0001",
+                "extension_b",
+                "/local/path/",
+                "full/file/name_key_B_0001.extension_b",
+            ],
+            [
+                "name_key_C_0002",
+                "extension_a",
+                "/local/path/",
+                "full/file/name_key_C_0002.extension_a",
+            ],
+            [
+                "name_key_C_0002",
+                "extension_b",
+                "s3://bucket-name",
+                "full/file/name_key_C_0002.extension_b",
+            ],
+        ]
+        manifest = pd.DataFrame(contents, columns=self.manifest_columns)
+        include_filters = ["*"]
+        exclude_filters = ["*A*", "*C*"]
+        expected_selection = {
+            "name_key_B_0000": {
+                "extension_a": {
+                    "location": "/local/path/",
+                    "key": "full/file/name_key_B_0000.extension_a",
+                },
+                "extension_b": {
+                    "location": "/local/path/",
+                    "key": "full/file/name_key_B_0000.extension_b",
+                },
+            },
+            "name_key_B_0001": {
+                "extension_a": {
+                    "location": "/local/path/",
+                    "key": "full/file/name_key_B_0001.extension_a",
+                },
+                "extension_b": {
+                    "location": "/local/path/",
+                    "key": "full/file/name_key_B_0001.extension_b",
+                },
+            },
+        }
+        actual_selection = filter_manifest_files.fn(
+            manifest=manifest,
+            extensions=self.extensions,
+            include_filters=include_filters,
+            exclude_filters=exclude_filters,
+        )
+        self.assertEqual(expected_selection, actual_selection)
+
+    def test_filter_manifest_files_include_B_exclude_0000(self):
```

Comment:
I think we should group these last two tests as subtests

Reply:
These two titles are hyper specific and don't really give much information about the API